### PR TITLE
Rotatable Wash Plant

### PR DESCRIPTION
**Pull request by MeiTianyou**
This allows the wash plant to be rotated. I think it looks pretty cool from the side if you cover it with glass.
wash_plant

I think it can look good using the flowing water for effect as well

rotate_for_aesthetics

### CODE REVIEW

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java [hunk 1]
```diff
@@ -138,8 +140,7 @@ public boolean checkMachine(IGregTechTileEntity aBaseMetaTileEntity, ItemStack a
 
     @Override
     protected IAlignmentLimits getInitialAlignmentLimits() {
-        // don't rotate a washer, water will flow out.
```
Comment (boubou19) [inline src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: this is exactly why it was not made rotatable in the first place.

**Review by boubou19 — CHANGES_REQUESTED** [review]

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java (hunk 1 above)
Comment (MeiTianyou) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: You can block off the "top" to stop the water flowing out. Besides that, letting the water flow out is also kind of cool looking, if you build a catchment for it.

Comment (boubou19) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: if you need extra handling when the multiblock is rotated, that's means it shouldn't being rotated.

It makes no sense that the washer should be rotated, as there is nothing happening in the water, you see nothing.

Comment (MeiTianyou) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: The extra handling to fill up the washer is there regardless of this change.

Comment (boubou19) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: did you know the washer is filling itself up?

Comment (MeiTianyou) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialWashPlant.java:141]: Yes, of course. That's what the changes in checkForWater are for, they make sure the multi fills itself up correctly even when rotated

**Comment by boubou19** [comment]
looks like Dream find this weird too, closing this.

**Comment by MeiTianyou** [comment]
Sad but OK, feel free to pull the checkForWater() refactor by itself if you like. I think it's a little clearer as to what's going on than the original.

**Pull request closed without merging**